skbio/sequence/_substitution.py

```diff
@@ -0,0 +1,518 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2013--, scikit-bio development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import numpy as np
+
+from skbio.util._decorator import experimental, classonlymethod
+from skbio.stats.distance import DissimilarityMatrix
+
+
+@experimental(as_of='0.5.10')
```

Comment:
Separate from this PR, I wonder about the long term value of these tags. Does anything use them?

Reply:
The documentation displays these notes. Back in days the team established a rigorous life cycle mechanism. See below a screenshot from this [video](https://www.youtube.com/watch?v=ZpgkRQooGqo). (I wasn't there so this is just my understanding.) I tend to this that this mechanism is beneficial. Other packages also have similar things.

![image](https://github.com/scikit-bio/scikit-bio/assets/4396343/80ea6f2a-f761-4454-9597-425b771d0b1a)

Reply:
What other packages are doing something like this? My concern is that while this did get setup, I don't recall seeing any PRs over the history of the project which actually changed things associated with these decorators nor am I aware of consumers of the library using this information (though I may be wrong there?)

Reply:
I echo your questions. Here are some examples in [Pandas](https://github.com/pandas-dev/pandas/blob/a671b5a8bf5dd13fb19f0e88edc679bc9e15c673/pandas/io/parsers/readers.py#L171) (and [here](https://github.com/pandas-dev/pandas/blob/a671b5a8bf5dd13fb19f0e88edc679bc9e15c673/pandas/core/frame.py#L12212)) and [scikit-learn](https://github.com/scikit-learn/scikit-learn/blob/3f89022fa/sklearn/decomposition/_pca.py#L199). Instead of adding decorators, they achieve this in the documentation only. We can have a discussion on how to deal with the decorators.

Reply:
Deprecation warnings, and noting versions added aren't a bad idea but qualititively different from these decorators

Reply:
I agree with you on this.